cantools/database/can/c_source.py

```diff
@@ -691,6 +834,15 @@ def maximum_value(self):
         else:
             return ((2 ** self.length) - 1)
 
+    def get_json_formatting(self):
+        # This returns a string that is used to format this signal
+        # into a JSON string.
+
+        if self.choices is not None:
+            return "\\\"%s:%g\\\""
+        # TODO Confirm float edge cases won't blow out the JSON output buffer.
```

Comment:
Should this have a precision limit on the float field? Otherwise binary representation issues may result in spurious digits.